From 727f9cd46fcc2047b867a59a9421d2d02e183766 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 9 Dec 2024 17:34:17 +0800
Subject: [PATCH 182/261] AOSCOS: gpio: gpio-nct6102: revise gpiochip_add() as
 gpiochip_add_data()

Per commit 3ff1180a39fb ("gpiolib: Remove data-less gpiochip_add()
function"), upstream maintainer advise against adding GPIO devices without
"driver-private data associated with the chip" but also recommended just
using NULL if none was specified.

Do the latter as recommended and revise the gpiochip_add() function as
gpiochip_add_data() with NULL as the second argument.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpio/gpio-nct6102.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-nct6102.c b/drivers/gpio/gpio-nct6102.c
index 1bb29ea5f5f2..143f0a139e21 100644
--- a/drivers/gpio/gpio-nct6102.c
+++ b/drivers/gpio/gpio-nct6102.c
@@ -177,6 +177,6 @@ static int __init nct6102_gpio_setup(void)
 	if (dev_id != SIO_NCT6102_ID)
 		return 0;
 
-	return gpiochip_add(&nct6102_chip);
+	return gpiochip_add_data(&nct6102_chip, NULL);
 }
 subsys_initcall(nct6102_gpio_setup);
-- 
2.48.1

